### DISCUSSION TITLE
Explore Image Embedding Drivers

### DISCUSSION THREAD

**Discussion by collindutter** [griptape-ai/griptape#1528] [Ideas]
- [x] I have read and agree to the [contributing guidelines](https://github.com/griptape-ai/griptape#contributing).

**Is your feature request related to a problem? Please describe.**
I can't use image embedding models like [OpenAi's CLIP](https://github.com/openai/CLIP) in Griptape.
**Describe the solution you'd like**
I want Embedding Drivers to support images.
**Describe alternatives you've considered**
- [fastembed](https://github.com/qdrant/fastembed)
- [transformers](https://huggingface.co/openai/clip-vit-base-patch32#use-with-transformers)

**Additional context**
The eventual goal is to use the embeddings with Vector Store Drivers which means I'd maybe need `upsert_image_artifact` and `query_image` methods. `upsert_vector` and [query_vector](https://github.com/griptape-ai/griptape/pull/1494) may be sufficient for now.

**Comment by collindutter** [comment]
Up for discussion: do we feel that this is in scope for Griptape? And if so, should we make the distinction between Text Embedding Drivers and Image Embedding Drivers?

**Comment by shhlife** [comment]
What would the workaround/process be if we didn't include image embedding drivers?

**Comment by collindutter** [comment]
Basically what is described [here](https://qdrant.tech/documentation/advanced-tutorials/multimodal-search-fastembed/), replacing the `QdrantClient` with `QdrantVectorStoreDriver`.

**Comment by shhlife** [comment]
Would we be able to do this with a LocalVectorStoreDriver as well? I feel like as long as we're doing text embeddings, and the future is multi-modal, we should handle image, audio, and video as well.

**Comment by shhlife** [comment]
I'm going to give it a try following the link you gave - but at a high level, I feel like we should offer the ability to embed things we have artifacts for.. at least common ones like images/audio/video especially as we go more multi-modal

**Comment by collindutter** [comment]
Tracking [here](https://github.com/griptape-ai/griptape/issues/1729)
